internal/monitors/processlist/processlist.go

```diff
@@ -0,0 +1,113 @@
+package processlist
+
+import (
+	"bytes"
+	"compress/zlib"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"math"
+	"runtime"
+	"time"
+
+	"github.com/signalfx/golib/event"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	monitorType = "processlist"
+	version     = "0.0.30"
+)
+
+// MONITOR(processlist): This monitor reports processlist information for Windows
+// Hosts.
+//
+// Sample YAML configuration:
+//
+// ```yaml
+// monitors:
+//  - type: processlist
+// ```
+
+var logger = log.WithFields(log.Fields{"monitorType": monitorType})
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `acceptsEndpoints:"false"`
+}
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+func compressBytes(in []byte) (buf bytes.Buffer, err error) {
+	compressor := zlib.NewWriter(&buf)
```

Comment:
This would probably make very little practical difference in the wild, but you can instantiate the zlib writer once and stick it on an instance field and then use the `Reset` method each read interval to avoid doing an allocation per read cycle.